zimui/src/stores/main.ts

```diff
@@ -1,52 +1,89 @@
 import { defineStore } from 'pinia'
-import axios from 'axios'
+import axios, { AxiosError }from 'axios'
 import Channel from '@/types/Channel'
 import Topic from '@/types/Topic'
 
 export type RootState = {
   channelData: Channel | null
   isLoading: boolean
   errorMessage: string
+  error: AxiosError<object> | null
 }
 export const useMainStore = defineStore('main', {
   state: () =>
     ({
       channelData: null,
       isLoading: false,
       errorMessage: '',
+      error: null,
     }) as RootState,
   getters: {},
   actions: {
     async fetchChannel() {
       this.isLoading = true
       this.errorMessage = ''
-      return axios.get('./channel.json').then(
-        (response) => {
-          this.isLoading = false
-          this.channelData = response.data as Channel
-        },
-        (_) => {
+      try {
+        const response = await axios.get('./channel.json')
```

Comment:
This is not correct, it does not return a promise anymore as it did before with `return axios.get(..).then(...)`. It is important to keep this async behavior. Error handling should be done within the promise context

Reply:
The statement itself is correct, just the return null was stopping the promise from getting returned. I checked it by logging it to the console, we are receiving the object as before.
![Screenshot from 2024-04-11 17-21-17](https://github.com/openzim/kolibri/assets/137636943/636acbe6-a222-44da-98ef-c83453bf260a)

Reply:
Asynk functions by default returns promises even though there is no explicit return statement. But your concern about chain promises is right. I have modified the code by removing the `await` and returning the promise as before.
Before committing all those changes I would like to hear from you about the error display styles I proposed over here
https://github.com/openzim/kolibri/pull/118#discussion_r1560901955